### DISCUSSION TITLE
Waypoint and Pictures

### DISCUSSION THREAD

**Discussion by Maggi68** [mastervanleeuwen/J-TrackGallery#47] [Q&A]
Hello,
this is a nice software here. 
First a question: It is only possible to load gpx data as a track not as waypoint? I got some error if I load a gpx file without a track.
I test it now and have read that their is a function to display locations of DP Calendar at the overview map.
So I asked me, is it also maybe principal possible to connect the Jtrackgallery with a picture Gallery to show the location of the stored pictures with Information of the picture in the map? For example in Joomgallery would the geolocation of the exif data and some IPTC data read out while the pictures would be uploaded and the data will be stored at the database. So if it possible to select a folder/folders from joomgallery to show all pictures at a map as waypoints would be fantastic. Maybe include a list of the title/Category and Exif/IPTX data. In Joomgallery it is only possible to show the location of one picture at the detailsite of the picutre not from all. 
Or a other question, if the attached pictures of an track in Jtrackgallery could be selected directly from the Joomgallery folders (not uploaded also new again), to attach some of the pictures to a track at the map and a click to the picture display it and will link to the Joomgallery detailpage.
Thanks and regards
Maggi

**Comment by mastervanleeuwen** [comment]
Hi Maggi,

Thanks for your message. A GPX file with waypoints only was not foreseen in the code, but I can try to see whether it can be adapted. Do you have an example of a file with waypoints only that you are trying to upload.

The idea of integrating Joomgallery support is an interesting one, but at the moment this is not foreseen, and I expect that it requires quite some coding. You can upload photos directly in J-TrackGallery and if they have geotagging information, they will be shown on the map.

Best regards, Marco.

**Comment by Maggi68** [comment]
Hello Marco,

thanks for your prompt answer...... I attached some gpx and KMZ files 
with or without pictures attached and will explain what I want to use 
for it...

About Joomgallery integrating, yes I can understand, this would be not 
so easy, but i am not a coder, so I dont know.......it was an idea that 
it would not necessary to store pictures also in Joomgallery and 
Jtrackgallery both as duplicate....and I am search a solution for my own 
workflow... let me explain...

I got a Table/list with waypoints (Name, descripton and Geolocation and 
other fields)

I load this Table and create a gpx file to import it into a app at 
mobile phone like TopoGPS or Locusmaps

If I arravie to the waypoint I take pictures from the waypoint as 
attached the pictures at the waypoint.

After I taken some pictures at different waypoints I export the 
waypoint(s) as KML or gpx with the pictures seperately or KMZ 
inclusively so I understand.

Now load the pictures to Joomgallery to show it. At the moment it is 
difficult to see which picture was taken at a specific waypoint after 
download it from a app. So I asked the developer of a app (TOPOGPS and 
Locus), to integrate a function like this: If I take a picture as a 
attachement at a waypoint, the name and description (like the 
geolocation at exif fields) will be stored in the picture at the IPTC 
fields (name and description and more is possible). They answer me, that 
they will look at this point it will be easier to find the pictures 
after export.... So if this will work or possible,  Joomgallery can read 
out the information stored at IPTC/EXIF in Software like Geosetter, 
Image Editing or also Joomgallery (since Version 3.4) can read out this 
information and put it at the database while upload the picture. But it 
is only possible to show the location of one picture and the information 
of the fields at detailpage for every single picture (not for all or 
selected folders). So I asked also Joomgallery Developer at github about 
the possibility to integrate the function to display a map for selected 
Categories/pictures from Joomgallery at a map with a list of the 
information (title, Description, Geoloctaion and so on) from the 
Exif/IPTCfields... in the meanwhile I saw your App and I thought it is 
maybe possible with a combination of teh apps with Joomgallery........

I dont know how complicated this is and how much time a coder 
need.....I'm also willing to pay a fee for it.

Thanks

Marcus
